contracts/scripts/foundry/InitializeL2ScrollOwner.s.sol

```diff
@@ -0,0 +1,237 @@
+    address L2_ERC1155_GATEWAY_PROXY_ADDR = vm.envAddress("L2_ERC1155_GATEWAY_PROXY_ADDR");
+
+    address L2_ETH_RATE_LIMITER_ADDR = vm.envAddress("L2_ETH_RATE_LIMITER_ADDR");
+    address L2_TOKEN_RATE_LIMITER_ADDR = vm.envAddress("L2_TOKEN_RATE_LIMITER_ADDR");
+
+    ScrollOwner owner;
+
+    function run() external {
+        vm.startBroadcast(L2_DEPLOYER_PRIVATE_KEY);
+
+        owner = ScrollOwner(payable(L2_SCROLL_OWNER_ADDR));
+
+        // @note we don't config 14D access, since the default admin is a 14D timelock which can access all methods.
+        configProxyAdmin();
+        configL1GasPriceOracle();
+        configL2TxFeeVault();
+        configWhitelist();
+        configL2ScrollMessenger();
+        configL2GatewayRouter();
+        configL2CustomERC20Gateway();
+        configL2ERC721Gateway();
+        configL2ERC1155Gateway();
+        configETHRateLimiter();
+        configTokenRateLimiter();
+
+        // @note comments out for testnet
+        // configL2USDCGateway();
+
+        grantRoles();
+        transferOwnership();
+
+        vm.stopBroadcast();
+    }
+
+    function transferOwnership() internal {
+        Ownable(L2_PROXY_ADMIN_ADDR).transferOwnership(address(owner));
+        Ownable(L1_GAS_PRICE_ORACLE_ADDR).transferOwnership(address(owner));
+        Ownable(L2_TX_FEE_VAULT_ADDR).transferOwnership(address(owner));
+        Ownable(L2_WHITELIST_ADDR).transferOwnership(address(owner));
+        Ownable(L2_SCROLL_MESSENGER_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L2_GATEWAY_ROUTER_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L2_CUSTOM_ERC20_GATEWAY_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L2_ETH_GATEWAY_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L2_STANDARD_ERC20_GATEWAY_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L2_WETH_GATEWAY_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L2_ERC721_GATEWAY_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L2_ERC1155_GATEWAY_PROXY_ADDR).transferOwnership(address(owner));
+
+        // Ownable(L2_USDC_GATEWAY_PROXY_ADDR).transferOwnership(address(owner));
+
+        Ownable(L2_ETH_RATE_LIMITER_ADDR).transferOwnership(address(owner));
+        AccessControlEnumerable(L2_TOKEN_RATE_LIMITER_ADDR).grantRole(bytes32(0), address(owner));
+        AccessControlEnumerable(L2_TOKEN_RATE_LIMITER_ADDR).revokeRole(bytes32(0), vm.addr(L2_DEPLOYER_PRIVATE_KEY));
+    }
+
+    function grantRoles() internal {
+        owner.grantRole(SECURITY_COUNCIL_NO_DELAY_ROLE, SECURITY_COUNCIL_ADDR);
+        owner.grantRole(SCROLL_MULTISIG_NO_DELAY_ROLE, SCROLL_MULTISIG_ADDR);
+        owner.grantRole(TIMELOCK_1DAY_DELAY_ROLE, L2_1D_TIMELOCK_ADDR);
+        owner.grantRole(TIMELOCK_7DAY_DELAY_ROLE, L2_7D_TIMELOCK_ADDR);
+
+        owner.grantRole(bytes32(0), L2_14D_TIMELOCK_ADDR);
+        owner.revokeRole(bytes32(0), vm.addr(L2_DEPLOYER_PRIVATE_KEY));
```

Comment:
```suggestion
        owner.grantRole(owner.DEFAULT_ADMIN_ROLE(), L2_14D_TIMELOCK_ADDR);
        owner.revokeRole(owner.DEFAULT_ADMIN_ROLE(), vm.addr(L2_DEPLOYER_PRIVATE_KEY));
```

Reply:
Fixed in [4830388](https://github.com/scroll-tech/scroll/pull/838/commits/48303885066cb8eb32574d69ed8e7001246e575c).